Gordon360/Services/HousingService.cs

```diff
@@ -31,39 +32,89 @@ public IEnumerable<HousingViewModel> GetAll()
         {
             return RawSqlQuery<HousingViewModel>.query("GET_STU_HOUSING_INFO");
         }
+        */
 
         /// <summary>
         /// Saves student housing info
+        /// - first, it creates a new row in the applications table and puts the id of the primary applicant and a timestamp
+        /// - second, it looks for the application id of the application with the information we just input (because 
+        /// the database creates the application ID so we have to ask it which number it generated for it)
+        /// - third, it inserts each applicant into the applicnts table along with the apartment ID so we know
+        /// which application on which they are an applicant
+        ///  
         /// </summary>
-        /// <returns>The housing item</returns>
-        public IEnumerable<HousingAppToSubmitViewModel>SaveApplication(string username, string sess_cde, string [] appIds)
+        /// <returns>Whether or not all the queries succeeded</returns>
+        public int SaveApplication(int apartAppId, string modifierId, string sess_cde, string [] applicantIds)
         {
-           //return RawSqlQuery<HousingViewModel>.query("INSERT_AA_APPLICATION", id);
-            IEnumerable<HousingAppToSubmitViewModel> result = null;
-            IEnumerable<HousingViewModel> result2 = null;
-            IEnumerable<HousingAppToSubmitViewModel> result3 = null;
-
+            IEnumerable<ApartmentAppSaveViewModel> result = null;
+            IEnumerable<ApartmentAppIDViewModel> idResult = null;
+            IEnumerable<ApartmentApplicantViewModel> result2 = null;
 
             DateTime now = System.DateTime.Now;
 
             var timeParam = new SqlParameter("@NOW", now);
-            var idParam = new SqlParameter("@MODIFIER_ID", username);
+            var modParam = new SqlParameter("@MODIFIER_ID", modifierId);
             var sessionParam = new SqlParameter("@SESS_CDE", sess_cde);
-            //var programParam = new SqlParameter("@APRT_PROGRAM", now);
 
-            result = RawSqlQuery<HousingAppToSubmitViewModel>.query("INSERT_AA_APPLICATION @NOW, @MODIFIER_ID", timeParam, idParam); //run stored procedure
-            result2 = RawSqlQuery<HousingViewModel>.query("GET_AA_APPID_BY_NAME_AND_DATE @NOW, @MODIFIER_ID", timeParam, idParam); //run stored procedure
-            foreach(string applicant in appIds){
-                result3 = RawSqlQuery<HousingAppToSubmitViewModel>.query("INSERT_AA_APPLICANT @APPLICATION_ID, @ID_NUM, @APRT_PROGRAM,  @SESS_CDE", result2, applicant, null, sessionParam); //run stored procedure
-            }
+            int appId = -1;
+
+            if (apartAppId == -1)
+            {
+                // If an application ID was not passed in, then check if an application already exists
+                idResult = RawSqlQuery<ApartmentAppIDViewModel>.query("GET_AA_APPID_BY_STU_ID_AND_SESS @SESS_CDE, @STUDENT_ID", sessionParam, modParam); //run stored procedure
```

Comment:
We should have the HousingController pass both the ID of the user who sent the request instead of having the modifierID passed in applicationDetails, so that this method of the housingService can compare the ID of the current user with the "ModifiedBy" ID number in the database before allowing the changes to be made. The  HousingController can has a way of getting the logged in user, as demonstrated in "GetApplicationID" in #440 , so using this would prevent spoofing the ID number of a different user.

I would then recommend making a separate endpoint dedicated to only handling the "change primary applicant", since that action requires passing a different user ID and modifying only the "ModifiedBy" value in the database entry. It would be much more efficient to put this in it's own endpoint rather than combining it with the full "save" operation. I will begin testing these ideas now

Reply:
Refer to #446 for a potential implementation to "change primary applicant"

Reply:
Looks good in #446.

Reply:
Looks good too in #446